lib/tests/modules/error-mkOption-in-config.nix

```diff
@@ -0,0 +1,10 @@
+{ lib, ... }:
+let
+  inherit (lib) mkOption;
+in
+{
+  wrong1 = mkOption {
+  };
+  nest.wrong2 = mkOption {
```

Comment:
While we're at it, it may also be reasonable to test that the correct error gets thrown for the nested case.

Reply:
Originally I believed this would affect the error, but the error doesn't go into this detail, and that might be for good reason, not to be tripped up by other errors. We'd have to parse those things as options to say anything about them, but that's making an assumption that they really are, and that this _config_ won't throw a worse exception on us.
For now the error condition is basically "you have configs but no options whatsoever" and the error message reflects that level of detail.

Reply:
> For now the error condition is basically "you have configs but no options whatsoever" and the error message reflects that level of detail.

Agreed. And I think considering that
* this option is already declared in this test fixture
* it's a slightly more special case since it's not at the top-level, but nested

it might be a good idea to check that the `It seems as if you.re trying to declare an option by placing it into .config. rather than .options` message also appears when trying to evaluate `config.nest.wrong2`.

Reply:
Done, and added comments to say that it currently throws for `config` anyway.